public/js/components/Editor.js

```diff
@@ -50,19 +87,37 @@ const Editor = React.createClass({
       ev => onWheel(this.editor, ev));
   },
 
-  onGutterClick(cm, line, gutter, ev) {
-    const bp = this.props.breakpoints.find(b => {
-      return b.getIn(["location", "line"]) === line + 1;
-    });
-
+  toggleBreakpoint(bp, line) {
     const applyBp = bp ? this.props.removeBreakpoint : this.props.addBreakpoint;
     applyBp({
       actor: this.props.selectedSource.get("actor"),
       line: line + 1,
-      snippet: cm.lineInfo(line).text.trim()
+      snippet: this.editor.lineInfo(line).text.trim()
     });
   },
 
+  onGutterClick(cm, line, gutter, ev) {
+    const bp = breakpointAtLine(this.props.breakpoints, line);
+    const location = {
+      actor: this.props.selectedSource.get("actor"),
+      line: line + 1
+    };
+
+    // show conditional breakpoint panel on command click
+    if (bp && ev.metaKey) {
+      const panel = renderConditionalBreakpointPanel({
+        location: location,
+        setBreakpointCondition: this.props.setBreakpointCondition,
+        closePanel: () => this.cbPanels[line].clear()
```

Comment:
You could make the panel get a reference to itself to close it, but you may be thinking that we want to close it from other places too which is why it's stored on the component, so that does make sense.